src/host/outputStream.cpp

```diff
@@ -368,27 +368,23 @@ void ConhostInternalGetSet::SetWorkingDirectory(const std::wstring_view /*uri*/)
 // Arguments:
 // - noteNumber - The MIDI note number to be played (0 - 127).
 // - velocity - The force with which the note should be played (0 - 127).
-// - duration - How long the note should be sustained (in milliseconds).
+// - duration - How long the note should be sustained.

```

Comment:
I personally disagree (but I'm open to changing my opinion).
I feel like there's times were specific types/units are warranted - for instance if you want to represent an exact 32-Bit integer you should use `uint32_t` - but there are also times were obfuscating the underlying type makes coding easier. Most programming languages for instance don't even have multiple duration types like C++. Instead they simply got a single `Duration` type or similar, which contains nanoseconds for instance. But even with C++ this is partially the same deal: `std::chrono::duration`'s underlying "ratio" is supposed to be transparent which is evident by how you can transparently convert any duration into any other duration and the copy operator will multiply/divide by the given underlying ratios.

Additionally, in this particularly case the duration we want to represent is neither in milliseconds, nor in microseconds at all: It's unit is in 1/32th of a second and we could choose to represent it in nanoseconds. Since `std::chrono::duration` obfuscates the unit of durations and allows duration conversions, saying that it's in `milliseconds` is kind of redundant IMO. It would make more sense in my opinion if the argument was a primitive integer for instance that didn't had the ratio builtin like the duration type does.

Reply:
I just want to make a point about the "1/32th of a second" thing. That's a limit of the `DECPS` sequence in particular, but my intention was to leave the API open to potentially being used for other operations as well. I chose microseconds as the unit because that seemed like a reasonable generic scale that would likely cover most use cases.